buildSrc/src/main/kotlin/com/yandex/practicum/middle_homework_5/gradle_plugins/FindUntranslatedStringsTask.kt

```diff
@@ -0,0 +1,66 @@
+package com.yandex.practicum.middle_homework_5.gradle_plugins
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import javax.xml.parsers.DocumentBuilderFactory
+import org.w3c.dom.Element
+import org.w3c.dom.Node
+
+abstract class FindUntranslatedStringsTask : DefaultTask() {
+    @TaskAction
+    fun findUntranslatedStrings() {
+        val untranslatedStrings = findAllUntranslatedStrings()
+
+        if (untranslatedStrings.isNotEmpty()) {
+            throw GradleException("missing localization for: $untranslatedStrings")
+        }
+    }
+
+    private fun findAllUntranslatedStrings(): Set<String> {
+        var strings: Set<String>? = null
+        val untranslatedStrings = mutableSetOf<String>()
+
+        File(project.projectDir, "src/main/res")
+            .listFiles { file ->
+                (file?.isDirectory ?: false) && file.name.startsWith("values")
```

Comment:
Элвис оператор и скобки можно убрать:
```suggestion
                file?.isDirectory == true && file.name.startsWith("values")
```